src/persist/heartbeat.test.ts

```diff
@@ -33,10 +34,7 @@ test('startHeartbeats starts interval that writes heartbeat each minute', async
       client2,
     }),
   );
-  await dagStore.withWrite(async (write: dag.Write) => {
-    await setClients(clientMap, write);
-    return write.commit();
-  });
+  await setClients(clientMap, dagStore);
```

Comment:
This test and similar ones for GC are now failing because the updating of the client state in the dagstore is no long synchronous on the clock.tick(ONE_MIN_IN_MS); (i.e. it is not done when the setInterval callback completes).  I'm not sure of a clean way to test this.

Reply:
isn't this what `await clock.tickAsync` etc is for? I use them in replicache.test.ts

Reply:
That didn't do it.  Seems tickAsync is for promises before the timers, not created by the timers.

https://sinonjs.org/releases/latest/fake-timers/
`The nextAsync() will also break the event loop, allowing any scheduled promise callbacks to execute before running the timers.`. 

I fixed this by exposing the promise to wait on.  Don't completely love it, but far cleaner than awaiting a setTimeout for X.